### ISSUE TITLE
Thumbnails are not loaded in 0.9.0

### ISSUE THREAD

**Issue by deep-jkl** [jsbroks/coco-annotator#170]
In recent update thumbnails stopped working. I believe that the frontend is not querying thumbnails from flask API, see outputs below. I see broken image icon in browser instead of thumbnails.

Previous version (0.7.6) output:

```
flask_1     |   do_handshake_on_connect: False
flask_1     |   ciphers: TLSv1
flask_1     |   raw_paste_global_conf: []
flask_1     | [2019-03-29 14:52:09 +0000] [6] [INFO] Starting gunicorn 19.9.0
flask_1     | [2019-03-29 14:52:09 +0000] [6] [DEBUG] Arbiter booted
flask_1     | [2019-03-29 14:52:09 +0000] [6] [INFO] Listening at: http://0.0.0.0:5000 (6)
flask_1     | [2019-03-29 14:52:09 +0000] [6] [INFO] Using worker: eventlet
flask_1     | [2019-03-29 14:52:09 +0000] [9] [INFO] Booting worker with pid: 9
flask_1     | [2019-03-29 14:52:09 +0000] [6] [DEBUG] 1 workers
flask_1     | [2019-03-29 14:52:15 +0000] [9] [DEBUG] GET /
flask_1     | [2019-03-29 14:52:15 +0000] [9] [DEBUG] GET /js/chunk-vendors.0272b52d.js
flask_1     | [2019-03-29 14:52:15 +0000] [9] [DEBUG] GET /css/app.46df1c76.css
flask_1     | [2019-03-29 14:52:15 +0000] [9] [DEBUG] GET /js/app.f9bc24ef.js
flask_1     | [2019-03-29 14:52:15 +0000] [9] [DEBUG] GET /socket.io/
flask_1     | [2019-03-29 14:52:15 +0000] [9] [DEBUG] GET /api/user/
flask_1     | [2019-03-29 14:52:15 +0000] [9] [DEBUG] GET /api/dataset/8/data
flask_1     | [2019-03-29 14:52:15 +0000] [9] [DEBUG] GET /api/info/
flask_1     | [2019-03-29 14:52:15 +0000] [9] [DEBUG] GET /socket.io/
flask_1     | [2019-03-29 14:52:15 +0000] [9] [DEBUG] GET /socket.io/
flask_1     | [2019-03-29 14:52:15 +0000] [9] [DEBUG] GET /favicon.ico
flask_1     | [2019-03-29 14:52:19 +0000] [9] [DEBUG] GET /api/image/2349/thumbnail
flask_1     | [2019-03-29 14:52:19 +0000] [9] [DEBUG] GET /api/image/2359/thumbnail
flask_1     | [2019-03-29 14:52:19 +0000] [9] [DEBUG] GET /api/image/2356/thumbnail
flask_1     | [2019-03-29 14:52:19 +0000] [9] [DEBUG] GET /api/image/2352/thumbnail
flask_1     | [2019-03-29 14:52:19 +0000] [9] [DEBUG] GET /api/image/2361/thumbnail
flask_1     | [2019-03-29 14:52:19 +0000] [9] [DEBUG] GET /api/image/2353/thumbnail
flask_1     | [2019-03-29 14:52:19 +0000] [9] [DEBUG] GET /api/image/2357/thumbnail
flask_1     | [2019-03-29 14:52:19 +0000] [9] [DEBUG] GET /api/image/2347/thumbnail
flask_1     | [2019-03-29 14:52:19 +0000] [9] [DEBUG] GET /api/image/2358/thumbnail
flask_1     | [2019-03-29 14:52:19 +0000] [9] [DEBUG] GET /api/image/2360/thumbnail
flask_1     | [2019-03-29 14:52:19 +0000] [9] [DEBUG] GET /api/image/2354/thumbnail
flask_1     | [2019-03-29 14:52:19 +0000] [9] [DEBUG] GET /api/image/2350/thumbnail
flask_1     | [2019-03-29 14:52:20 +0000] [9] [DEBUG] GET /api/image/2346/thumbnail
flask_1     | [2019-03-29 14:52:20 +0000] [9] [DEBUG] GET /api/image/2348/thumbnail
flask_1     | [2019-03-29 14:52:20 +0000] [9] [DEBUG] GET /api/image/2355/thumbnail
flask_1     | [2019-03-29 14:52:20 +0000] [9] [DEBUG] GET /api/image/2351/thumbnail
flask_1     | [2019-03-29 14:52:32 +0000] [9] [DEBUG] GET /api/image/2346
flask_1     | [2019-03-29 14:52:32 +0000] [9] [DEBUG] GET /api/annotator/data/2346
flask_1     | [2019-03-29 14:52:41 +0000] [9] [DEBUG] POST /api/annotator/data
flask_1     | [2019-03-29 14:52:41 +0000] [9] [DEBUG] GET /api/image/2347
flask_1     | [2019-03-29 14:52:41 +0000] [9] [DEBUG] GET /api/annotator/data/2347
```

version 0.9.0:
```
Starting annotator_mongodb ... done
Recreating coco-annotator_flask_1 ... done
Attaching to annotator_mongodb, coco-annotator_flask_1
flask_1     | [2019-03-29 14:55:23 +0000] [6] [INFO] Starting gunicorn 19.9.0
flask_1     | [2019-03-29 14:55:23 +0000] [6] [INFO] Listening at: http://0.0.0.0:5000 (6)
flask_1     | [2019-03-29 14:55:23 +0000] [6] [INFO] Using worker: eventlet
flask_1     | [2019-03-29 14:55:23 +0000] [9] [INFO] Booting worker with pid: 9
flask_1     | [2019-03-29 14:55:24 +0000] [9] [WARNING] MaskRCNN model is disabled.
flask_1     | [2019-03-29 14:55:24 +0000] [9] [WARNING] DEXTR model is disabled.
flask_1     | [2019-03-29 14:55:26 +0000] [9] [INFO] Socket connection created with jkl
flask_1     | [2019-03-29 14:55:31 +0000] [9] [INFO] Socket connection has been disconnected with jkl
flask_1     | [2019-03-29 14:55:31 +0000] [9] [INFO] Socket connection created with jkl
flask_1     | [2019-03-29 14:55:31 +0000] [9] [INFO] jkl has started annotating image 2347
flask_1     | [2019-03-29 14:55:33 +0000] [9] [INFO] Socket connection has been disconnected with jkl
flask_1     | [2019-03-29 14:55:33 +0000] [9] [INFO] Socket connection created with jkl
flask_1     | [2019-03-29 14:55:33 +0000] [9] [INFO] jkl has started annotating image 2347
```

**Comment by jsbroks** [comment]
Did you upgrade from 0.7.6 to 0.9? Did you clear the cache in the browser? the thumbnail endpoint has been removed

**Comment by deep-jkl** [comment]
I have figured it out, images are mounted on read-only filesystem, and the flask API cannot create thumbnails folder, therefore it fails. I suggest to create a fallback option for case that dataset folder is not writable, what do you think?

**Comment by jsbroks** [comment]
If you want to put a PR for this feature go for it. The follow back could be to just regenerate it every time

**Comment by jsbroks** [comment]
This issue wont be fix, since as of 0.10.3 write access to the datasets folder is required for exports.

**Comment by a-jahani** [comment]
I fixed it with ```sudo chmod -R 777 <PATH_TO_DATASET>``` giving write permission to all folders for all users.
